server/server.js

```diff
@@ -1,65 +1,106 @@
+  let id = req.params.reservationid;
+  client.get(id, (err, result) => {
+    if(result) {
+      res.send(JSON.parse(result));
+    } else {
+      db.getReservationByReservationId(req.params, (err, result) => {
+        if (err) {
+          res.status(500).send({err: `Server oopsie ${err}`});
+        } else {
+          client.setex(id, 60, JSON.stringify(result));
+          res.send(result);
+        }
+      });
+    }
+  })
+});
 
 app.get('/api/listings/:listingId', (req, res) => {
+  var id = req.params.listingId;
 
-  db.getListingById(req.params, (err, result) => {
-    if (err) {
-      res.status(500).send({ err: `Server oopsie ${err}` });
-    } else if (result.length === 0) {
-      res.status(404).send('No such listing')
+  client.get(id, (err, result) => {
+    if(result) {
+      res.send(JSON.parse(result));
     } else {
-      db.getReviewsByListingId(result[0].review_id, (err, reviews) => {
-        if (err) {
-          res.status(500).send({err: `Server oopsie ${err}`})
-        } else {
-          result[0].reviews = reviews[0];
-          res.send(result[0]);
+      db.getListingById(req.params, (err, result) => {
+      if (err) {
+        res.status(500).send({ err: `Server oopsie ${err}` });
+      } else if (result.length === 0) {
+        res.status(404).send('No such listing')
+      } else {
+        db.getReviewsByListingId(result[0].review_id, (err, reviews) => {
+            if (err) {
+              res.status(500).send({err: `Server oopsie ${err}`})
+            } else {
+              result[0].reviews = reviews[0];
+              client.setex(id, 60, JSON.stringify(result[0]));
+              res.send(result[0]);
+            }
+          });
         }
-      })
+      });
     }
   });
-
 });
 
 app.get('/api/dates/:listingId', (req, res) => {
   // TODO: refactor using router
   let method = db.getBookedDatesByListingId;
-  let data = null; 
+  let data = null;
 
-  if (req.query.targetDate) {
+  if(req.query.targetDate) {
     method = db.getFirstBookedDateAfterTarget;
-    let target = req.query.targetDate.split('-');
+    const target = req.query.targetDate.split('-');
     data = [req.params.listingId, ...target];
   }
 
-  if (req.query.month) {
-    let month = req.query.month.split('-');
+  if(req.query.month) {
+    const month = req.query.month.split('-');
     data = [req.params.listingId, ...month];
   }
 
-  method(data, (err, result) => {
-    if (err) {
-      res.status(500).send({ err: `Server oopsie ${err}` });
-    } else res.send(result);
-  });
+  let id = req.params.listingId;
 
+  client.get(id, (err, resu) => {
+    if(resu) {
+      res.send(JSON.parse(resu));
+    } else {
+      method(data, (err, result) => {
+        if(err) {
+          res.status(500).send({err: `Server oopsie ${err}`});
+        } else {
+          //console.log('here');
+          client.setex(id, 60, JSON.stringify(result));
+          res.send(result);
+        }
+      });
+    }
+  });
 });
 
```

Comment:
Do you have the other CRUD apis? I didn't see put or delete, although I did see the POST todo.